src/common/buffer.cc

```diff
@@ -1945,7 +1945,8 @@ int buffer::list::write_fd(int fd) const
     return write_fd_zero_copy(fd);
 
   // use writev!
-  iovec iov[IOV_MAX];
+  int max = MIN(_buffers.size(), IOV_MAX);
```

Comment:
@dachary  Sorry for the delayed reply.  It seems that the caller should guarantee the overflow issue. Shall we fix this like:

``` C++
int max = _buffers.size();
assert(max <= IOV_MAX);
```